src/Nethermind/Nethermind.State/PartialStorageProviderBase.cs

```diff
@@ -290,7 +289,7 @@ public virtual void ClearStorage(Address address)
         /// </summary>
         protected class Change
         {
-            public Change(ChangeType changeType, StorageCell storageCell, byte[] value)
+            public Change(ChangeType changeType, in StorageCell storageCell, in UInt256 value)
```

Comment:
Have you checked how this works and whether there's no benefit to pass by value? I vaguely remember that in some cases it was nicely JITted, but I don't recall whether a class or a struct it was.